### PR TITLE
Change hover message in decile chart

### LINKED ISSUE #431: Add baseline and new income to decile impact chart hovercards
Currently it's the plotly default of, e.g.
>Decile=1
Relative change=2.1%

All other charts on the main page have sentences as hover cards, so let's add that to this too, e.g.:
>Households in the 1st decile gain an average of {£157, 2.1%} per year, from £10,000 to £10,157

### PR DESCRIPTION
**Pull request by tolaouk**
fixes #431

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Could you try `git pull upstream main` and syncing? Per https://github.com/isaacs/github/issues/750#issuecomment-852713101 that should fix the excessive files changed from creating this branch from the in-progress one.

**Comment by MaxGhenis** [comment]
`make debug-client` is failing with this:
>openfisca_core.errors.cycle_error.CycleError: Circular definition detected on formula in_poverty_bhc@2022

Here's the [full error message](https://gist.github.com/MaxGhenis/52990907634889f6eeab99ba93369528).

This PR doesn't change poverty formulas. @nikhilwoodruff any ideas?

**Comment by MaxGhenis** [comment]
@tolaouk could you check if the `make debug-client` command works when you check out the master branch?

**Comment by tolaouk** [comment]
> @tolaouk could you check if the `make debug-client` command works when you check out the master branch?

@MaxGhenis It works with master branch.

File: policyengine/impact/population/charts.py [hunk 1]
```diff
@@ -31,38 +31,93 @@ def decile_chart(
...
-        reformed.calc(config.household_net_income_variable)
-        - baseline_household_net_income
+        reform_household_net_income - baseline_household_net_income
     )
     household_size = baseline.calc("people", map_to="household")
     # Group households in decile such that each decile has the same
     # number of people
     baseline_household_equiv_income.weights *= household_size
     household_decile = baseline_household_equiv_income.decile_rank()
+    agg_gain_by_decile = household_gain.groupby(household_decile).sum()
```
Comment (MaxGhenis) [inline policyengine/impact/population/charts.py:45]: On second thought, we could simplify this and avoid the `pd.Series(...).reset_index(drop=True)` statements by using DataFrames more. Something like this (though `mdf.MicroDataFrame` might simplify):
```python
# Create household DataFrame to sum.
w = baseline.weights
household = pd.DataFrame(dict(
    decile = baseline_household_equiv_income.decile_rank(),
    baseline_agg_income = baseline.calc(
        config.household_net_income_variable
    ) * w,
    reform_agg_income = reformed.calc(
        config.household_net_income_variable
    ) * w,
    households = w,
))
# Sum household dataframe by decile.
decile = household.groupby(decile).sum().reset_index()
# Calculate relevant ratios.
decile["baseline_income_per_hh"] = decile.baseline_agg_income / decile.households
# ...
# Create labels.
decile["verb"] = np.where(
        decile.mean_gain > 0,
        "rise",
        np.where(mean_gain < 0, "fall", "remain"),
    )
# ...
```
Then you can just pass the `decile` DataFrame to the chart at the end.

**Comment by nikhilwoodruff** [comment]
@MaxGhenis does the server definitely fail? I've oddly seen that error a few times, but it has still provided a response.

**Comment by MaxGhenis** [comment]
Oh not sure, it was on @tolaouk's machine. Tola could you try it and see if it still works?

**Review by MaxGhenis — APPROVED** [review]

File: policyengine/impact/population/charts.py (hunk 1 above)
Comment (MaxGhenis) [reply policyengine/impact/population/charts.py:83]: I'll file an issue to refactor separately.

**Pull request merged**